src/mono/mono/mini/method-to-ir.c

```diff
@@ -1101,6 +1101,9 @@ type_from_op (MonoCompile *cfg, MonoInst *ins, MonoInst *src1, MonoInst *src2)
 		case STACK_I8:
 			ins->opcode = OP_LCONV_TO_R_UN; 
 			break;
+		case STACK_R4:
```

Comment:
This should also fix https://github.com/dotnet/runtime/issues/34382 (I arrived here just now while looking into this). My only feedback here is that it'd be nice to have a `g_assert_not_reached ()` default case.